### ISSUE TITLE
introduce backup lock

### ISSUE THREAD

**Issue by rompetroll** [mimiro-io/datahub#143]
As extra safety mechanism we want to make sure that datahub's backup manager does not accidently overwrite backup locations with the wrong content.

proposed mechanism:

- upon startup a new datahub can create a “storage-id” and put that into a file in the storage location. 
- During successive starts, datahub detects the storage-id file and does not overwrite it
- backup sync must also copy the “storage-id” file to the backup location.
- before start, backup checks whether there is a “storage-id” in the backup location.
- backup is only executed if there is no “storage-id” yet in backup location, or if the “storage-id” file content matches the running datahub’s “storage-id” value.
- mismatches are logged as error and must be dealt with manually

TLDR;: introduce a sync id file.
